layers/gpu/instrumentation/gpu_shader_instrumentor.cpp

```diff
@@ -392,27 +426,28 @@ void GpuShaderInstrumentor::PreCallRecordCreatePipelineLayout(VkDevice device, c
                                                               const VkAllocationCallbacks *pAllocator,
                                                               VkPipelineLayout *pPipelineLayout, const RecordObject &record_obj,
                                                               chassis::CreatePipelineLayout &chassis_state) {
-    if (chassis_state.modified_create_info.setLayoutCount >= adjusted_max_desc_sets_) {
+    if (chassis_state.modified_create_info.setLayoutCount > desc_set_bind_index_) {
         std::ostringstream strm;
-        strm << "Pipeline Layout conflict with validation's descriptor set at slot " << desc_set_bind_index_ << ". "
-             << "Application has too many descriptor sets in the pipeline layout to continue with gpu validation. "
-             << "Validation is not modifying the pipeline layout. "
-             << "Instrumented shaders are replaced with non-instrumented shaders.";
-        InternalError(device, record_obj.location, strm.str().c_str());
+        strm
+            << "pCreateInfo::setLayoutCount (" << chassis_state.modified_create_info.setLayoutCount
+            << ") will conflicts with validation's descriptor set at slot " << desc_set_bind_index_ << ". "
+            << "Application has too many descriptor sets in the Pipeline Layout to continue with GPU Shader Instrumentation. "
+            << "Validation is not modifying the Pipeline Layout and instrumented SPIR-V are replaced with non-instrumented SPIR-V.";
```

Comment:
The message should be update then, to clearly state that all pipelines created with the faulty layout will not be instrumented, but only those. Something like:
```suggestion
            << "pCreateInfo::setLayoutCount (" << chassis_state.modified_create_info.setLayoutCount
            << ") will conflicts with validation's descriptor set at slot " << desc_set_bind_index_ << ". "
            << "This pipeline layout will not allow GPU shader instrumentation to be setup for pipelines created with it, so no error error will be repored for them by GPU-AV."
```

Reply:
yep updated